kernel/frame_allocator/src/lib.rs

```diff
@@ -469,10 +472,8 @@ impl<const S: MemoryState> Drop for Frames<S> {
             MemoryState::Free => {
                 if self.size_in_frames() == 0 { return; }
         
-                let mut free_frames: FreeFrames = Frames {
-                    typ: self.typ,
-                    frames: self.frames.clone(),
-                };
+                let frames = core::mem::replace(&mut self.frames, FrameRange::empty());  
```

Comment:
we have to replace the `FrameRange` here rather than the whole `Frames` because trying to replace the `Frames` object leads to a compiler error for mismatched types, the dest is &mut Frames<S> and the src is a FreeFrames.

Reply:
got it. Hope that's not too hard to verify, seems tough but it works for me here.